src/NetMQ/IReceivingSocket.cs

```diff
@@ -1,18 +1,62 @@
     /// </summary>
     public interface IReceivingSocket
     {
         /// <summary>
-        /// Read an available message into the given Msg. The message content is in the form of a byte-array
-        /// which higher-level methods may convert into a string.
+        /// Read message data into the given <see cref="Msg"/> object.
         /// </summary>
-        /// <param name="msg">a Msg struct to receive the message data into</param>
-        /// <param name="options">a SendReceiveOptions value that can specify the DontWait or SendMore bits</param>
+        /// <remarks>
+        /// If a message is available, its content will be transferred to <paramref name="msg"/> and the call will
+        /// return immediately.
+        /// <para/>
+        /// If no message is available, the behaviour of this method depends upon <paramref name="options"/>:
+        /// <list type="bullet">
+        ///   <item>
+        ///     <see cref="SendReceiveOptions.DontWait"/> — causes the method to immediately throw
+        ///     <see cref="AgainException"/>.
+        ///   </item>
+        ///   <item>
+        ///     <see cref="SendReceiveOptions.None"/> — blocks until the next message arrives, and cannot be
+        ///     interrupted. This a convenient and safe when you know a message is available, such as for code
+        ///     within a <see cref="NetMQSocket.ReceiveReady"/> callback.
+        ///     <para/>
+        ///     <see cref="NetMQSocket"/> objects having non-negative <see cref="SocketOptions.ReceiveTimeout"/>values
+        ///     will timeout if a message is not received within the specified number of milliseconds.
+        ///   </item>
+        /// </list>
+        /// </remarks>
+        /// <param name="msg">An object to receive the message's data into.</param>
+        /// <param name="options">A set of flags that control receipt. Only <see cref="SendReceiveOptions.None"/> and
+        /// <see cref="SendReceiveOptions.DontWait"/> have any effect. <see cref="SendReceiveOptions.SendMore"/> makes 
+        /// no sense here and is ignored.</param>
+        /// <exception cref="AgainException">No message was available within the allowed timeout period. Try again.
+        /// </exception>
+        [Obsolete("Use Receive(ref Msg) or TryReceive(ref Msg,TimeSpan) instead.")]
         void Receive(ref Msg msg, SendReceiveOptions options);
+        
+        /// <summary>
+        /// Block until the next message arrives, then make the message's data available via <paramref name="msg"/>.
+        /// </summary>
+        /// <remarks>
+        /// The call  blocks until the next message arrives, and cannot be interrupted. This a convenient and safe when
+        /// you know a message is available, such as for code within a <see cref="NetMQSocket.ReceiveReady"/> callback.
+        /// </remarks>
+        /// <param name="msg">An object to receive the message's data into.</param>
+        void Receive(ref Msg msg);
```

Comment:
I like this proposal. It is simpler. I was thinking that there may be some future optimisation for the indefinite blocking case, but in practice it could just a single branch per receive call.